lib/ObjWriter/objwriter.cpp

```diff
@@ -386,30 +388,53 @@ GetSymbolRefExpr(ObjectWriter *OW, const char *SymbolName,
   return MCSymbolRefExpr::create(T, Kind, OutContext);
 }
 
-extern "C" void EmitSymbolRef(ObjectWriter *OW, const char *SymbolName,
-                              int Size, bool IsPCRelative, int Delta = 0) {
+enum class RelocType
+{
+    IMAGE_REL_BASED_ABSOLUTE    = 0x00,
+    IMAGE_REL_BASED_HIGHLOW     = 0x03,
+    IMAGE_REL_BASED_DIR64       = 0x0A,
+    IMAGE_REL_BASED_REL32       = 0x10,
+};
+
+extern "C" int EmitSymbolRef(ObjectWriter *OW, const char *SymbolName,
+                             RelocType RelocType, int Delta) {
   assert(OW && "ObjWriter is null");
   auto *AsmPrinter = &OW->getAsmPrinter();
   auto &OST = static_cast<MCObjectStreamer &>(*AsmPrinter->OutStreamer);
   MCContext &OutContext = OST.getContext();
 
-  // Get symbol reference expression
-  const MCExpr *TargetExpr = GetSymbolRefExpr(OW, SymbolName);
+  bool IsPCRelative = false;
+  int Size = 0;
+  MCSymbolRefExpr::VariantKind Kind = MCSymbolRefExpr::VK_None;
 
-  switch (Size) {
-  case 8:
-    assert(!IsPCRelative && "NYI no support for 8 byte pc-relative");
+  // Convert RelocType to MCSymbolRefExpr
+  switch (RelocType) {
+  case RelocType::IMAGE_REL_BASED_ABSOLUTE:
+    Kind = MCSymbolRefExpr::VK_COFF_IMGREL32;
```

Comment:
Once we start using this then it implies Windows/COFF only. Can you add an assertion like this?:
```
assert(OW->MOFI->getObjectFileType() == OW->MOFI->IsCOFF);
```
I guess we will minimize such use for Window specific layout/representation.